source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2664 @@
+===================
+Unified Test Format
+===================
+
+:Spec Title: Unified Test Format
+:Spec Version: 1.0.0
+:Author: Jeremy Mikola
+:Advisors: Prashant Mital, Isabel Atkinson, Thomas Reggi
+:Status: Draft
+:Type: Standards
+:Minimum Server Version: N/A
+:Last Modified: 2020-09-15
+
+.. contents::
+
+--------
+
+Abstract
+========
+
+This project defines a unified schema for YAML and JSON specification tests,
+which run operations against a MongoDB deployment. By conforming various spec
+tests to a single schema, drivers can implement a single test runner to execute
+acceptance tests for multiple specifications, thereby reducing maintenance of
+existing specs and implementation time for new specifications.
+
+
+META
+====
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in `RFC 2119 <https://www.ietf.org/rfc/rfc2119.txt>`__.
+
+
+Goals
+=====
+
+This test format can be used to define tests for the following specifications:
+
+- `Change Streams <../change-streams/change-streams.rst>`__
+- `Command Monitoring <../command-monitoring/command-monitoring.rst>`__
+- `CRUD <../crud/crud.rst>`__
+- `GridFS <../gridfs/gridfs-spec.rst>`__
+- `Retryable Reads <../retryable-reads/retryable-reads.rst>`__
+- `Retryable Writes <../retryable-writes/retryable-writes.rst>`__
+- `Sessions <../sessions/driver-sessions.rst>`__
+- `Transactions <../transactions/transactions.rst>`__
+- `Convenient API for Transactions <../transactions-convenient-api/transactions-convenient-api.rst>`__
+
+This is not an exhaustive list. Specifications that are known to not be
+supported by this format may be discussed under `Future Work`_.
+
+
+Specification
+=============
+
+
+Terms
+-----
+
+Entity
+  Any object or value that is indexed by a unique name and stored in the
+  `Entity Map`_. This will typically be a driver object (e.g. client, session)
+  defined in `createEntities`_ but may also be a
+  `saved operation result <operation_saveResultAsEntity_>`_. Entities are
+  referenced throughout the test file (e.g. `Entity Test Operations`_).
+
+Internal MongoClient
+  A MongoClient created specifically for use with internal test operations, such
+  as inserting collection data before a test, configuring fail points during a
+  test, or asserting collection data after a test.
+
+
+Schema Version
+--------------
+
+This specification and the `Test Format`_ follow
+`semantic versioning <https://semver.org/>`__. The version is primarily used to
+validate test files with a `JSON schema <https://json-schema.org/>`__ and also
+allow test runners to determine whether a particular test file is supported.
+
+New tests files SHOULD always be written using the latest major version of this
+specification; however, test files SHOULD be conservative in the minor version
+they specify (as noted in `schemaVersion`_).
```

Comment:
@saghm This and the note in schemaVersion relate to your earlier comments about not defaulting to the most recent schema version. I specify that the latest major should be used, but preference should be given to the most conservative choice for minor.